test_cases/scripts/run_all_categories.sh

```
@@ -0,0 +1,100 @@
+
+    echo -e "${BLUE}=== $display_name ===${NC}"
+    echo "$description"
+    echo ""
+
+    TOTAL_CATEGORIES=$((TOTAL_CATEGORIES + 1))
+
+    if cd "../$directory" && ./"$script"; then
+        echo -e "${GREEN}✅ $display_name PASSED${NC}"
+        CATEGORY_RESULTS+=("$display_name: ✅ PASSED")
+        PASSED_CATEGORIES=$((PASSED_CATEGORIES + 1))
+        cd - >/dev/null
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)